### DISCUSSION TITLE
accessing plugins' "actions" from python

### DISCUSSION THREAD

**Discussion by bdcht** [NationalSecurityAgency/ghidra#4244] [Q&A] [answered]
Hi,
is it possible to trigger "actions" that are defined in some plugins (and usually triggered by the user through the GUI) from a python script ? More generally, is it possible from python to get a reference to a chosen Plugin object (say the DataTypeManagerPlugin for example) ?

For example:
is there a way to trigger the "Commit To Archive" action available in the DataTypeManager using only the python API ?

If not, I am trying to "replicate" what the action source code does in my own script, but here I need to access or instanciate a DataTypeSynchronizer, which requires a DataTypeManagerHandler instance as 1st argument...which I could get if I had a ref on the plugin but I don't.

**Comment by dragonmacher** [comment, marked as answer]
The action API is not designed for that.  It is possible, but quite clunky and not for the uninitiated. 

Alternatively, each action usually calls into an existing API method that you can call directly.   So, you can use the source code to figure out what the action does and then you *usually* can call the same code in your script.  Admittedly, this is not as convenient as simply calling an existing action.

**Reply by bdcht** [reply]
Thanks for the reply,
will look into replicating what the DataTypeSynchronizer.commit does.
